src/NServiceBus.TransportTests/When_scope_dispose_throws.cs

```diff
@@ -37,7 +36,44 @@ public class When_scope_dispose_throws : NServiceBusTransportTest
 
             // since some transports doesn't have native retry counters we can't expect the attempts to be fully consistent since if
             // dispose throws the message might be picked up before the counter is incremented
-            Assert.LessOrEqual(1, errorContext.ImmediateProcessingFailures);
+            Assert.LessOrEqual(2, errorContext.ImmediateProcessingFailures);
+        }
+
+        [Test]
+        public async Task Should_properly_increment_immediate_processing_failures()
```

Comment:
regarding its own file: I started down that route and found that the file should be named "When_scope_dispose_throws". Am I wrong there?

the TestCase TransportTransactionMode stuff you're talking about are the attributes right?